openedx/core/lib/xblock_utils.py

```diff
@@ -446,3 +448,23 @@ def _course_info_content(html_parsed):
                         return payload
 
     return course_update_items
+
+
+def xblock_local_resource_url(block, uri):
+    """
+    Returns the URL for an XBlock's local resource.
+
+    Note: when running with the full Django pipeline, the file will be accessed
+    as a static asset which will use a CDN in production.
+    """
+    xblock_class = block.__class__.unmixed_class
```

Comment:
Nit: it might be worth using `getattr(block.__class__, 'unmixed_class', block.__class__)`, to handle the off chance that mixins aren't enabled.